ocaml/runtime/domain.c

```diff
@@ -770,6 +754,7 @@ static void domain_create(uintnat initial_minor_heap_wsize,
   domain_state->trap_barrier_block = -1;
 #endif
 
+  caml_reset_young_limit(domain_state);
```

Comment:
This line should not be there. It was removed by f4098874dfd045eb53634f01f3763f3b815ed3aa ("Minor changes to complete statmemprof"), which is not in 5.2 but was backported as part of #2461

Reply:
Fixed